mitxgraders/comparers/comparers.py

```diff
@@ -178,3 +179,126 @@ def eigenvector_comparer(comparer_params_evals, student_eval, utils):
         }
 
     return utils.within_tolerance(actual, expected)
+
+def vector_span_comparer(comparer_params_evals, student_eval, utils):
+    """
+    Check whether student's answer is nonzero and in the span of some given
+    vectors.
+
+    comparer_params: A list of vectors
+
+    Usage
+    =====
+
+    >>> from mitxgraders import MatrixGrader
+    >>> grader = MatrixGrader(
+    ...     answers={
+    ...         'comparer_params': [
+    ...             '[3, x, 1 + i]',
+    ...         ],
+    ...         'comparer': vector_span_comparer
+    ...     },
+    ...     variables=['x'],
+    ... )
+    >>> grader(None, '[3, x, 1 + i]')['ok']
+    True
+    >>> grader(None, '[9, 3*x, 3 + 3*i]')['ok']
+    True
+    >>> grader(None, '[9, 3*x, 3 - 3*i]')['ok']
+    False
+
+    Complex scale factors work, too:
+    >>> grader(None, '(4 + 2*i)*[3, x, 1 + i]')['ok']
+    True
+
+    Student input should be nonzero:
+    >>> grader(None, '[0, 0, 0]')['ok']
+    False
+
+    Input shape is validated:
+    >>> grader(None, '5')
+    Traceback (most recent call last):
+    InputTypeError: Expected answer to be a vector, but input is a scalar
+
+    Multiple vectors can be provided:
+    >>> grader = MatrixGrader(
+    ...     answers={
+    ...         'comparer_params': [
+    ...             '[1, 1, 0]',    # v0
+    ...             '[0, 1, 2]'     # v1
+    ...         ],
+    ...         'comparer': vector_span_comparer
+    ...     },
+    ... )
+
+    Above, 2*v0 + 3i*v1:
+    >>> grader(None, '[2, 2 + 3*i, 6*i]')['ok']
+    True
+    """
+
+    # All comparer_param_evals should have same shape, or numpy.linalg.lstsq
+    # will let us know.
+    utils.validate_shape(student_eval, comparer_params_evals[0].shape)
+
+    if utils.within_tolerance(0, np.linalg.norm(student_eval)):
+        return {
```

Comment:
Shouldn't this just return False?

Reply:
You mean instead of the `{ok, msg, grade}` dict? If I remember correctly comparer function is similar to edx check function and can return either True/False or a `{ok, msg, grade}` dict.

(Although, I just realized I updated the comment at top of file to say only True/False. That was incorrect.)

Reply:
I think there's no particular reason to return the dict here, so just return False. You are correct though that you can return the dict.

Reply:
Ah, ignore me - I see that you're returning a message :-)